solo-io/gloo
# Fix istioIntegration.enabled, only use istioIntegration.enabled for Istio automtls

# Description
- Fixes `9-gateway-proxy-configmap.yaml` to add `gateway_proxy_sds` if `istioIntegration.enabled=true` is set 
- Removes `istioSDS.enabled` from settings. Setting deprecated `istioSDS.enabled` will no longer be sufficient to enable Istio automtls integration. 

## API changes
No API changes. Updates proto descriptions to require `istioIntegration.enabled=true` instead of `istioSDS` in settings and automtls plugin

## Code changes
- Updates`9-gateway-proxy-configmap.yaml` to add check if `istioIntegration.enabled=true` is set
- Removes `istioSDS.enabled` from settings. Setting deprecated `istioSDS.enabled` will no longer be sufficient to enable Istio automtls integration. You will need to set `istioIntegration.enabled=true` to enable autotmls.  

## CI changes
None

## Docs changes
- Updates descriptions to require `istioIntegration.enabled=true` instead of `istioSDS` in settings and automtls plugin

# Context
Slack thread: https://solo-io-corp.slack.com/archives/C03MFATU265/p1717613043976389?thread_ts=1717536053.313579&cid=C03MFATU265 

## Interesting decisions
None

## Testing steps
- Istio e2e regression tests confirm old helm values work as expected 
- New kubernetes e2e test confirm Istio automtls works
- Added helm tests for deprecated `istioSDS.enabled` value

## Notes for reviewers
None 

# Checklist:
